packages/stark-ui/src/modules/table/components/table.component.spec.ts

```diff
@@ -1105,4 +1103,36 @@ describe("TableComponent", () => {
 			expect(component.selectChanged.emit).toHaveBeenCalledWith([dummyData[0]]);
 		});
 	});
+
+	describe("async", () => {
+		const dummyData: object[] = [
+			{ id: 1, description: "dummy 1" },
+			{ id: 2, description: "dummy 2" },
+			{
+				id: 3,
+				description: "dummy 3"
+			}
+		];
+		console.log(dummyData);
+
+		beforeEach(() => {
+			hostComponent.columnProperties = [{ name: "id" }, { name: "description" }];
+			hostComponent.dummyData = undefined; // data starts uninitialized
+
+			hostFixture.detectChanges(); // trigger data binding
+			component.ngAfterViewInit();
+		});
+
+		it("should update rows after async data fetch", () => {
+			const rowsBeforeData: NodeListOf<HTMLTableRowElement> = hostFixture.nativeElement.querySelectorAll("table tbody tr");
+			expect(rowsBeforeData.length).toBe(0);
+
+			// "async fetch of data resolves"
+			hostComponent.dummyData = dummyData;
+			hostFixture.detectChanges();
+
+			const rowsAfterData: NodeListOf<HTMLTableRowElement> = hostFixture.nativeElement.querySelectorAll("table tbody tr");
+			expect(rowsAfterData.length).toBe(3);
```

Comment:
Maybe it is better to put `dummyData.length` instead of `3` to make it more "dynamic" so this doesn't need to change if we change the dummy data at some point ;)